packages/main/src/Table.js

```diff
@@ -654,73 +711,173 @@ class Table extends UI5Element {
 	 * Handles Alt + Up/Down.
 	 * Switches focus between column header, last focused item, and "More" button (if applicable).
 	 * @private
-	 * @param {boolean} shouldMoveUp Whether to move focus upward
-	 * @param {object} focusedElement The element currently in focus
+	 * @param {CustomEvent} event
 	 */
-	_handleArrowAlt(shouldMoveUp, focusedElement) {
-		const focusedElementType = this.getFocusedElementType(focusedElement);
-		const moreButton = this.getMoreButton();
+	_handleArrowAlt(event) {
+		const shouldMoveUp = isUpAlt(event);
+		const focusedElementType = this.getFocusedElementType(event.target);
 
 		if (shouldMoveUp) {
 			switch (focusedElementType) {
 			case "tableRow":
-				this._previousFocusedRow = focusedElement;
-				return this._onColumnHeaderClick();
+			case "tableGroupRow":
+				this._prevFocusedRow = event.target;
+				return this._onColumnHeaderClick(event);
 			case "columnHeader":
-				return moreButton ? moreButton.focus() : this._previousFocusedRow.focus();
+				return this.morеBtn ? this.morеBtn.focus() : this._prevFocusedRow.focus();
 			case "moreButton":
-				return this._previousFocusedRow ? this._previousFocusedRow.focus() : this._onColumnHeaderClick();
+				return this._prevFocusedRow ? this._prevFocusedRow.focus() : this._onColumnHeaderClick(event);
 			}
 		} else {
 			switch (focusedElementType) {
 			case "tableRow":
-				this._previousFocusedRow = focusedElement;
-				return moreButton ? moreButton.focus() : this._onColumnHeaderClick();
+			case "tableGroupRow":
+				this._prevFocusedRow = event.target;
+				return this.morеBtn ? this.morеBtn.focus() : this._onColumnHeaderClick(event);
 			case "columnHeader":
-				if (this._previousFocusedRow) {
-					return this._previousFocusedRow.focus();
-				}
-
-				if (moreButton) {
-					return moreButton.focus();
+				if (this._prevFocusedRow) {
+					this._prevFocusedRow.focus();
+				} else if (this.morеBtn) {
+					this.morеBtn.focus();
 				}
 
 				return;
 			case "moreButton":
-				return this._onColumnHeaderClick();
+				return this._onColumnHeaderClick(event);
 			}
```

Comment:
I missed the fact that there is already the same here (as the same structure " if <-> switch ") already in the code. I would leave it up to you but may be it would be best to have them in the same way. So if you decide to leave the previous structure for the _handleTab method as well I am ok with that.

Reply:
Here it wouldn’t be sufficient as there are more checks than just `shouldMoveUp`, and it has been previously refactored https://github.com/SAP/ui5-webcomponents/pull/4550#discussion_r786796866